### ISSUE TITLE
What would it take to update to a newer .NET version?

### ISSUE THREAD

**Issue by markski1** [ikkentim/SampSharp#438]
Hello, my question is basically the title. Imagine I wanted to update SampSharp for targetting .NET 8, what would need to be changed? Would it require a rewrite of the library or minor things?

It catches my eye that the default Project package targets .NET Core 3.1, despite this library not seeming abandoned, so I was wondering how viable would it be to try and bring it up to date

**Comment by markski1** [comment]
Hi Ciprian, thanks for the quick answer!

I haven't tried this yet because I don't have a Linux machine available, but theoretically, if I targetted .NET 8, and put the x32 Linux runtime of .NET 8 in my "runtime" folder for my samp server, would it then run just fine on .NET 8?

**Comment by CiprianN23** [comment]
Yes, it will just be fine

**Comment by LYZENCORE-9** [comment]
SampSharp is supporting .NET 8 , i test it and it's working

**Comment by markski1** [comment]
Many thanks

**Comment by markski1** [comment]
> SampSharp is supporting .NET 8 , i test it and it's working

Sorry to bring this up again, but may I ask, where did you find an x86 build of .NET 8? Seems the official download is only x64

**Comment by LYZENCORE-9** [comment]
> > SampSharp is supporting .NET 8 , i test it and it's working
> 
> Sorry to bring this up again, but may I ask, where did you find an x86 build of .NET 8? Seems the official download is only x64

go to this repository [https://github.com/Servarr/dotnet-linux-x86](NET 8 x86_64 Linux Unofficial Build)

**Comment by markski1** [comment]
Very nice, thanks once again.

**Comment by markski1** [comment]
@MrDave1999 I ran into the same problem and sadly did not have enough free time to troubleshoot it. It seems to balloon in commit memory size until it crashes out. Very strange

**Comment by MrDave1999** [comment]
Yes, that's the problem: https://github.com/Servarr/dotnet-linux-x86/pull/5#issuecomment-1968268268

**Comment by LYZENCORE-9** [comment]
> @LYZENCORE-9 @markski1 I can't run dotnet linux x86 for .net 8 (https://github.com/Servarr/dotnet-linux-x86/releases/tag/v8.0.4-90). I'm testing it on a VPS with ubuntu 20.04 but it does not run gamemode:
> 
> ![image](https://private-user-images.githubusercontent.com/43916038/379935628-640095fa-38cf-43b1-b89b-c0fc94b98286.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzAwNTMwODcsIm5iZiI6MTczMDA1Mjc4NywicGF0aCI6Ii80MzkxNjAzOC8zNzk5MzU2MjgtNjQwMDk1ZmEtMzhjZi00M2IxLWI4OWItYzBmYzk0Yjk4Mjg2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDI3VDE4MTMwN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU4M2M0MGQxMTc3NTRhNDlkYmZlOWIwMzEwNTJkMjU1ODQxYjZkZThkMzY4NGYxMzIwMzFhMmU3ODA2ZDhmOWYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.JYGt2DX_DaNcFgxxbf3_3mZPAnKOfeWtlV9KqPvM1Ok)
> 
> As shown in the image, it keeps initializing the runtime and from there it does nothing, the server never starts. Do you know what it could be?

weird , i use this runtimes in my server at my home and it works

**Comment by LYZENCORE-9** [comment]
if you want to create a good and stable server , just create your own multiplayer mod or create it with Zeex's SAMPGDK/SDK

**Comment by KoertLichtendonk** [comment]
Also running into this issue where it is initializing the runtime, consumes all memory briefly and then kills the process. Any known solutions would be greatly appreciated!

**Comment by MrDave1999** [comment]
@KoertLichtendonk I have not found any solution. The only thing we could do is to manually compile the runtime .net 8.0 for an x86 machine.
For now I am running my server with this runtime: https://github.com/Servarr/dotnet-linux-x86/releases/tag/v6.0.35-97

**Comment by Duduzera1997** [comment]
Based on my recent experiences, using updated versions of .NET is only possible on Windows servers. Servers running Linux unfortunately still need to use .NET 6 with the [https://github.com/Servarr/dotnet-linux-x86/releases/tag/v6.0.35-97](https://github.com/Servarr/dotnet-linux-x86/releases/tag/v6.0.35-97) runtime.

I think the one way to solve this problem once and for all would be to use Open.mp in 64 bits and use .NET x64. I don't know if this is possible yet, but I see that finding a .NET 8 x86 runtime that works would only be a pothole filler because we would have the problem happening with each new .NET version, so we would always need a compilation of the working x86 runtime. 

Yes, this is very boring, Microsoft unfortunately does not support 32-bit Linux so we will not have x86 being provided by them officially.

**Comment by LYZENCORE-9** [comment]
@Duduzera1997 @MrDave1999 
after i finish from some works and some tests from my academy , i will recompile all .NET 8 runtimes for 32-bit linux target (x86)

**Comment by NeekoGta** [comment]
Hello, did you found a solution? 
I manually compiled version 8.0.23 but I have the same problem than you @DevD4v3.

``` 
[2026-01-22T12:48:06+0000] [Info] [SampSharp:INFO] CoreCLR: runtime/shared/Microsoft.NETCore.App/8.0.23/libcoreclr.so
[2026-01-22T12:48:06+0000] [Info] [SampSharp:INFO] Initializing .NET runtime...
[2026-01-22T12:48:28+0000] [Info] [SampSharp:ERROR] Failed to initialize CoreCLR runtime. Error -2147024882.
```

You can try: https://drive.google.com/file/d/151KqfPCpNmkh8Gaf33uKYcEngx_Je5l0/view?usp=sharing
